### PR TITLE
Fix: Fix nodeRequest method

### PR DESCRIPTION
**Pull request by timcubb**
The nodeRequest method in retrieve.js does not currently work - it returns once we have the first chunk, instead of waiting for all of the chunks to be collected. 

This fix is needed to be able to use the SDK in Node.

### CODE REVIEW

File: src/retrieve.js [hunk 1]
```diff
@@ -120,13 +120,14 @@ function nodeRequest(options) {
...
   return MiniPromise.createPromise(function(resolve, reject) {
+    let response = '';
     const parts = URL_PATTERN.exec(options.url);
     if (!parts) {
       throw new Error('Evolv: Invalid endpoint URL');
     }
 
     const schema = parts[1];
-    self && (schema === 'http' ? self.import('http') : self.import('https')).then(function (http) {
+    (schema === 'http' ? import('http') : import('https')).then(function (http) {
```
[the remark this answers is not in the document]
Comment (timcubb) [reply src/retrieve.js:130]: @mattstrom - self is always undefined and throwing errors for me

[the remark this answers is not in the document]
Comment (mattstrom) [reply src/retrieve.js:130]: Is it necessary to remove for your changes?

[the remark this answers is not in the document]
Comment (timcubb) [reply src/retrieve.js:130]: @mattstrom - yeah, I could not get it to work with `self`

[the remark this answers is not in the document]
Comment (robertsevernsentient) [reply src/retrieve.js:130]: @timcubb -- I think we need this:
void !function () {
  typeof self === 'undefined' && typeof global === 'object'
    ? global.self = global : null
}()

From this library - https://www.npmjs.com/package/node-self

**Comment by robertsevernsentient** [comment]
@timcubb -- what's needed to progress this? Is this change required?

**Comment by mattstrom** [comment]
Closing. I'm addressing this in a forthcoming PR.

**Pull request closed without merging**